File: python_backend/game_logic/fenParser/getBoard/convert_id.py


def convert_id(fen_id):
    """ fenParser task 4, convert fen id to a game id"""
    if fen_id.isupper():
        return 'W' + fen_id
    else:
        return 'B' + fen_id.upper()


def convert_ids(pos_list):
    """convert each fen_id in the pos_list to a game id"""

    for i in range(len(pos_list)):
        if len(pos_list[i]) == 1 and pos_list[i].isalpha():
            pos_list[i] = convert_id(pos_list[i])

    return pos_list


if __name__ == "__main__":

    # test 1:
    print('test 1:')
    pos_list = convert_ids( ['R', 'N', 'B', 'Q', 'K', 'B', 'N', 'R', 'P', 'P', 'P', 
                             'P', 'P', 'P', 'P', 'P', '#', '#', '#', '#', '#', '#', 
                             '#', '#', '#', '#', '#', '#', '#', '#', '#', '#', '#', 
                             '#', '#', '#', '#', '#', '#', '#', '#', '#', '#', '#', 
                             '#', '#', '#', '#', 'p', 'p', 'p', 'p', 'p', 'p', 'p', 
                             'p', 'r', 'n', 'b', 'q', 'k', 'b', 'n', 'r'])

    print(pos_list)
    print("\n")

    # test 2
    print('test 2')
    pos_list = convert_ids( ['#', 'R', 'L', 'T', 'K', '#', '#', '#', '#', '#', '#',
                             '#', '#', '#', '#', '#', 'R', 'N', 'B', 'Q', '#', 'B', 
                             'N', 'R', '#', '#', '#', '#', '#', '#', '#', '#', 'r', 
                             'n', 'b', 'q', '#', 'b', 'n', 'r', '#', '#', '#', 'k', 
                             '#', '#', '#', '#', '#', '#', '#', '#', '#', '#', '#', 
                             '#', '#', '#', '#', '#', '#', '#', '#', '#'])
    print(pos_list)
    print("\n")

File: app (1)/src/game_logic/fenParser/getBoard/convertId.js
import {isalpha} from "../../helpers/isalpha";
import {isupper} from "../../helpers/isupper";


export function convertId(fenId) {
    /* fenParser task 4, convert fen id to a game id**/
    if (isupper(fenId)) {
        return 'W' + fenId
    }
    else {
        return 'B' + fenId.toUpperCase()
    }
}


export function convertIds(posList) {
    /*convert each the.includes(fenId) posList to a game id**/
    for (let i = 0; i < posList.length; i++) {
        if (posList[i].length === 1 && isalpha(posList[i])) {
            posList[i] = convertId(posList[i])
        }
    }
    return posList
}

// module.exports = convertIds;

/** for node.js
if (require.main === module) {
    // test 1:
    console.log('test 1:')
    let posList = convertIds( ['R', 'N', 'B', 'Q', 'K', 'B', 'N', 'R', 'P', 'P', 'P', 
                             'P', 'P', 'P', 'P', 'P', '#', '#', '#', '#', '#', '#', 
                             '#', '#', '#', '#', '#', '#', '#', '#', '#', '#', '#', 
                             '#', '#', '#', '#', '#', '#', '#', '#', '#', '#', '#', 
                             '#', '#', '#', '#', 'p', 'p', 'p', 'p', 'p', 'p', 'p', 
                             'p', 'r', 'n', 'b', 'q', 'k', 'b', 'n', 'r'])
    console.log(posList)
    console.log("\n")

    // test 2
    console.log('test 2')
    posList = convertIds( ['#', 'R', 'L', 'T', 'K', '#', '#', '#', '#', '#', '#',
                             '#', '#', '#', '#', '#', 'R', 'N', 'B', 'Q', '#', 'B', 
                             'N', 'R', '#', '#', '#', '#', '#', '#', '#', '#', 'r', 
                             'n', 'b', 'q', '#', 'b', 'n', 'r', '#', '#', '#', 'k', 
                             '#', '#', '#', '#', '#', '#', '#', '#', '#', '#', '#', 
                             '#', '#', '#', '#', '#', '#', '#', '#', '#'])
    console.log(posList)
    console.log("\n")

}
*/